freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -355,13 +368,54 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
         }
     }
 
+    private SyncStatusDetail internalSynchronizeStackForDeleteUser(Stack stack, String deletedWorkloadUser) {
```

Comment:
could you move this to a separate class and refactor into multiple shorter methods

Reply:
I've made it slightly shorter. We can do additional refactoring when working on the new sync endpoints (as mention in a previous comment), so I prefer to keep it in this class for now.